core/src/toga/app.py

```diff
@@ -391,6 +391,16 @@ def home_page(self) -> str | None:
         menu items."""
         return self._home_page
 
+    @property
+    def dark_mode(self) -> bool | None:
+        """Whether the user has dark mode enabled in their environment (read-only).
+
+        Returns None if we cannot determine dark mode settings."""
+        try:
+            return self._impl.get_dark_mode_state()
+        except NotImplementedError:
+            return None
```

Comment:
If the underlying dark mode implementations use `factory.not_implemented()`, this can be simplified to a direct `self._impl.get_dark_mode_state()`